### ISSUE TITLE
lazyPopulate's leafCheck default implementation not fully functional

### ISSUE THREAD

**Issue by sannies** [edvin/tornadofx#424]
The default leafCheck impl executes invokeChildFactorySynchronously() and sets the childFactoryInvoked flag. Subsequent calls to getChildren then never execute the code that actually sets the children

```
  override fun getChildren(): ObservableList<TreeItem<T>> {
        if (!childFactoryInvoked) {
            task {
                invokeChildFactorySynchronously()
            } success {
                if (childFactoryResult != null) {
                    super.getChildren().setAll(childFactoryResult!!.map { newLazyTreeItem(it) })
                    listenForChanges()
                }
            }
        }
        return super.getChildren()
    }
```

**Comment by sannies** [comment]
I obviously workarounded by implementing a proper leafCheck therefore no big prio for me. Nevertheless I'd like to safe the next one's time by filing the ticket.

**Comment by edvin** [comment]
Thank you! Would you have time to submit a PR?

**Comment by nimakro** [comment]
I am afraid I introduced the bug with commit 76735c57baa1bd759a24f96725d4e1c50a2f922b I submitted a fix (#469)  feel free to have a look.

**Comment by edvin** [comment]
Great work @nimakro - I've merged now.
